### PR TITLE
[CHANGED] Use interface instead of struct for micro.Request

### PR DESCRIPTION
**Pull request by piotrpio**
- changed `micro.Request` to an interface
- moved tests to test package

### CODE REVIEW

File: micro/request.go [hunk 1]
```diff
@@ -25,16 +25,41 @@ type (
...
+		// Headers returns request headers.
+		Headers() Headers
+
+		// Subject returns underlying NATS message subject.
+		Subject() string
 	}
 
 	// RequestHandler is a function used as a Handler for a service.
-	RequestHandler func(*Request)
+	RequestHandler func(Request)
```
Comment (wallyqs) [inline micro/request.go:53]: Handler should probably be interface based here too a la http.Handler, that would make it convenient to pass around types that can handle requests, think something like this:
```go
// micro.Handler
type Handler func(Request)
func (fn RequestHandler) Handle(req Request) {
  fn(req)
}
```

File: micro/example_test.go [hunk 2]
```diff
@@ -28,33 +29,33 @@ func ExampleAddService() {
...
 
-	echoHandler := func(req *Request) {
+	echoHandler := func(req micro.Request) {
 		req.Respond(req.Data())
 	}
 
-	config := Config{
+	config := micro.Config{
 		Name:        "EchoService",
 		Version:     "v1.0.0",
```
Comment (wallyqs) [inline micro/example_test.go:38]: this should be `1.0.0` or would get a validation error, maybe improve the regex to allow for either? Some packages that parse semver allow for either style too

Comment (Jarema) [reply micro/example_test.go:38]: We agreed to use the official semver regex that does not allow for "v" in front, it not being part of semantic version.

File: micro/request.go (hunk 1 above)
Comment (piotrpio) [reply micro/request.go:53]: Very good idea, thanks!

File: micro/example_test.go (hunk 2 above)
Comment (piotrpio) [reply micro/example_test.go:38]: Yes, exactly. In this case, the example is broken, I'll fix it.

File: micro/request.go (hunk 1 above)
Comment (piotrpio) [reply micro/request.go]: added

File: micro/request.go [hunk 3]
```diff
@@ -22,19 +22,54 @@ import (
 )
 
 type (
+	// Handler is used to respond to service requests.
+	Handler interface {
+		Handle(Request)
+	}
+
+	// HandlerFunc is a function implementing [Handler].
+	// It allows using a function as a reqeust hadnler, without having to implement Handle
```
Comment (wallyqs) [inline micro/request.go:31]:
```suggestion
	// It allows using a function as a request handler, without having to implement Handle
```

**Review by wallyqs — APPROVED** [review]
LGTM 👍

**Pull request merged**